_posts/2021-01-13-#19825.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: pr
+date: 2020-01-13
+title: "Simpler setban and new ban manipulation commands"
+pr: 19825
+authors: [dhruv]
+components: ["p2p", "rpc/rest/zmq"]
+host: dhruv
+status: upcoming
+commit: e650bba
+---
+
+## Notes
+
+- `BanMan` has two functions:
+    1. Banning: Manually configured by the user via the `setban` rpc. No
+       outbound/inbound connections will be established with a banned address or
+       subnet; their address will not be gossiped. Banned addresses are
+       persisted in `banlist.dat` on shutdown and reloaded on startup.
+    2. Discouragement: [Misbehaving
+       peers](https://github.com/bitcoin/bitcoin/blob/86a8b35f/src/net_processing.cpp#L1018)
+       are discouraged over time. Inbound connections to discouraged peers are
+       allowed but preferred for eviction. Outbound connections are not made and
+       their addresses are not gossiped.
+
+- Neither banning nor discouragement protect against DoS attacks as IP
+  addresses are inexpensive.
+
+- Programatic banning or disconnection increase the risk of a network split.
+
+- The `BanMan` interface allows banning and unbanning via `CNetAddr` and
+  `CSubNet`. This PR consolidates to only accept `CSubNet` (an IP address is a
+  subnet of 1).
+
+- This PR also adds two new RPC commands:
+    1. `setban listbanned ip`: lists all the ban entries that include `ip`
+    2. `setban removeall ip`: removes all ban entries that include `ip`
+
+## Questions
+
+1.  What are some use cases for banning specific IP addresses or subnets?
+
+2.  Is programatic banning/disconnection an effective DoS countermeasure for
+    misbehaving peers?
+
+3.  How can programatic banning/disconnection be abused by an attacker?
```

Comment:
s/programatic/programmatic|automatic/ here and in 2 other places

Reply:
I agree that automatic is better.